semeio/jobs/scripts/spearman_correlation.py

```diff
@@ -1,19 +1,36 @@
 import argparse
 
+from ert_data.measured import MeasuredData
 from ert_shared.libres_facade import LibresFacade
 from res.enkf import ErtScript
-
+from semeio.jobs.correlated_observations_scaling.exceptions import EmptyDatasetException
+from semeio.jobs.scripts.correlated_observations_scaling import (
+    CorrelatedObservationsScalingJob,
+)
 from semeio.jobs.spearman_correlation_job.job import spearman_job
 
 
 class SpearmanCorrelationJob(ErtScript):
     def run(self, *args):
         facade = LibresFacade(self.ert())
 
+        obs_keys = [
+            facade.get_observation_key(nr)
+            for nr, _ in enumerate(facade.get_observations())
+        ]
+        measured_data = MeasuredData(facade, obs_keys)
+
         parser = spearman_job_parser()
         args = parser.parse_args(args)
 
-        spearman_job(facade, args.threshold, args.dry_run)
+        scaling_configs = spearman_job(measured_data, args.threshold)
+
+        if not args.dry_run:
+            for config in scaling_configs:
```

Comment:
I belive we could also pass a single configuration listing the different scalings that are to be done. Any particular reason you choose to pass them one by one?

Reply:
No, just went with what was there. Good call.

Reply:
Wait, are you talking about multiple configs inside one config? I'd rather not introduce that as it seems buggy (ref https://github.com/equinor/semeio/issues/41).

I don't have to do a generator though, will return a list of configs.

Reply:
>  I'd rather not introduce that as it seems buggy

You would not introduce it, as the job already supports it (at least in the yaml, but not in if passed as a programmatic object). I think we should either utilise it or remove the feature...

Reply:
I have nothing against the generator ;)

Reply:
It is already used by a site, but okay, let's move to multiple configs in a config.

(Aside) Any reason for not using yaml streams here?

Reply:
Or we revert to only doing a single one in the same config?